Makefile

```
@@ -297,120 +311,137 @@ $(O_FILES): | asset_files
 
 .PHONY: o_files asset_files
 
+
 #### Main commands ####
 
 ## Cleaning ##
 clean:
-	$(RM) -rf $(ROMC) $(ROM) $(ELF) build
+	$(RM) -r $(BUILD_DIR)
 
 assetclean:
-	$(RM) -rf $(ASSET_BIN_DIRS)
-	$(RM) -rf assets/text/*.h
-	$(RM) -rf build/assets
-	$(RM) -rf .extracted-assets.json
+	$(RM) -r $(ASSET_BIN_DIRS)
+	$(RM) -r $(BUILD_DIR)/assets
+	$(RM) -r assets/text/*.h
+	$(RM) -r .extracted-assets.json
 
 distclean: assetclean clean
-	$(RM) -rf asm baserom data
+	$(RM) -r asm data
+	$(RM) -r baseroms/$(VERSION)/segments
 	$(MAKE) -C tools clean
 
+venv:
+	test -d $(VENV) || python3 -m venv $(VENV)
+	$(PYTHON) -m pip install -U pip
+	$(PYTHON) -m pip install -U -r requirements.txt
+
 ## Extraction step
-setup:
+setup: venv
 	$(MAKE) -C tools
```

As mentioned in discord, I think the `setup` step shouldn't setup the venv and update the dependencies each time it is run.

```suggestion
setup:
	$(MAKE) -C tools
```